Modern/ops/src/test/java/org/bonej/ops/ellipsoid/DistanceFromEllipsoidSurfaceOpTest.java

```diff
@@ -0,0 +1,147 @@
+
+
+/**
+ * Tests for {@link DistanceFromEllipsoidSurfaceOp}.
+ *
+ * @author Alessandro Felder
+ */
+public class DistanceFromEllipsoidSurfaceOpTest {
+
+    private static final ImageJ IMAGE_J = new ImageJ();
+    private static BinaryFunctionOp<Ellipsoid, Point3d, DoubleType> distanceFromEllipsoidSurfaceOp;
+    private static UnitSphereRandomVectorGenerator sphereRng = new UnitSphereRandomVectorGenerator(3);
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        distanceFromEllipsoidSurfaceOp = Functions.binary(IMAGE_J.op(), DistanceFromEllipsoidSurfaceOp.class, DoubleType.class, Ellipsoid.class, Point3d.class);
+    }
+
+    @Test
+    public void testDistanceOnAxesForAxisAlignedEllipsoid(){
+
+        Ellipsoid ellipsoid = new Ellipsoid(1.0, 2.0, 3.0);
+
+        Point3d outsidePoint  = new Point3d(2.0,0.0,0.0);
+        Point3d insidePoint  = new Point3d(0.0,1.0,0.0);
+        Point3d surfacePoint  = new Point3d(0.0,0.0,3.0);
+
+        double distanceToOutsidePoint = distanceFromEllipsoidSurfaceOp.calculate(ellipsoid,outsidePoint).get();
+        double distanceToInsidePoint = distanceFromEllipsoidSurfaceOp.calculate(ellipsoid,insidePoint).get();
+        double distanceToSurfacePoint = distanceFromEllipsoidSurfaceOp.calculate(ellipsoid,surfacePoint).get();
+
+        assertEquals("Distance to outside point failed",1.0,distanceToOutsidePoint,1.0e-12);
+        assertEquals("Distance to inside point failed",1.0,distanceToInsidePoint,1.0e-12);
+        assertEquals("Distance to surface point failed",0.0,distanceToSurfacePoint,1.0e-12);
+
+    }
+
+    //Ellipsoid and points from previous test translated by (1,1,1) and rotated 30 degrees around x-axis
+    @Test
+    public void testDistanceOnAxesForArbitraryEllipsoid(){
+
+        Ellipsoid ellipsoid = new Ellipsoid(1.0, 2.0, 3.0);
+        ellipsoid.setOrientation(new Matrix3d(1,0,0,0,Math.sqrt(3.0)/2.0,0.5,0,-0.5,Math.sqrt(3.0)/2.0));
+        ellipsoid.setCentroid(new Vector3d(1,1,1));
+
+        Point3d outsidePoint  = new Point3d(1.0+2.0,1.0,1.0);
+        Point3d insidePoint  = new Point3d(1.0,1.0+Math.sqrt(3.0)/2.0,1-0.5);
+        Point3d surfacePoint  = new Point3d(1.0,1.0+1.5,1.0+3.0*Math.sqrt(3.0)/2.0);
+
+        double distanceToOutsidePoint = distanceFromEllipsoidSurfaceOp.calculate(ellipsoid,outsidePoint).get();
+        double distanceToInsidePoint = distanceFromEllipsoidSurfaceOp.calculate(ellipsoid,insidePoint).get();
+        double distanceToSurfacePoint = distanceFromEllipsoidSurfaceOp.calculate(ellipsoid,surfacePoint).get();
+
+        assertEquals("Distance to outside point failed",1.0,distanceToOutsidePoint,1.0e-12);
+        assertEquals("Distance to inside point failed",1.0,distanceToInsidePoint,1.0e-12);
+        assertEquals("Distance to surface point failed",0.0,distanceToSurfacePoint,1.0e-12);
+
+    }
+
+    @Test(expected = ArithmeticException.class)
+    public void testDistanceOriginToUnitSphere() throws Exception
+    {
```

Comment:
Actually tests that method throws ArithmeticException if determinant zero, so name should reflect that.